arcade/types.py

```diff
@@ -410,6 +412,324 @@ def random(
         return cls(r, g, b, a)
 
 
+class ColorFloat(RGBANormalized):
```

Comment:
Is this a good candidate for `NamedTuple`?  I think it gives us the `@property`s, the deepcopy, and the default `a` value for free.

I used this snippet to test, I may have missed corner cases.

```python
class Color(NamedTuple):
    r: float
    g: float
    b: float
    a: float = 1.0

a = Color(1, 2, 3)
b = copy.deepcopy(a)
print(repr(a), repr(b))
print(a is b)
```

Reply:
Yeah, I see einarf saying NamedTuple is slow, but not saying that what you have written is faster.  That makes me think that both might be equally slow.  Did anyone benchmark at the time of that conversation?

There's good precedent for this question, since one of my first benchmarks found that shapely was slower even though the shapely code was meant to be faster.